src/hardhat.ts

```diff
@@ -0,0 +1,68 @@
+import { CHAINS } from './generated/chains';
+import { toUpperSnakeCase } from './utils/strings';
+import { Chain, HardhatEtherscanConfig, HardhatNetworksConfig } from './types';
+
+export function buildEnvVariables(): string[] {
+  return CHAINS
+    .filter((chain) => chain.explorer?.api?.key?.required)
+    .map((chain) => buildEtherscanApiKeyName(chain));
+}
+
+export function buildEtherscanApiKeyName(chain: Chain): string {
+  return `${toUpperSnakeCase(chain.alias)}_ETHERSCAN_API_KEY`;
+}
+
+// https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-verify#multiple-api-keys-and-alternative-block-explorers
+export function buildEtherscanConfig(): HardhatEtherscanConfig {
+  // Not usable outside of a Node.js environment
+  if (typeof window !== 'undefined') {
+    return { apiKey: {}, customChains: [] };
```

Comment:
Should you not throw instead?

Reply:
Probably yeah. Running outside of a Node environment is wrong